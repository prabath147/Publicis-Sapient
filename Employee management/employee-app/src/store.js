import { createStore, applyMiddleware } from "redux";
import { composeWithDevTools } from "redux-devtools-extension";
import rootReducer from "./Component/reducers";
import Thunk from "redux-thunk";

const store = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(Thunk))
);

export default store;
