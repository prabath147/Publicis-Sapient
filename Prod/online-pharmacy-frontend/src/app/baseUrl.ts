export const baseURL = "https://frk5iphcfk.execute-api.us-east-1.amazonaws.com/";
    // "http://localhost:8081";
    
