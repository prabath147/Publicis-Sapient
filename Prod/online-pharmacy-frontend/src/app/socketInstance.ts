
export const socketURL = "https://c8zpvs7fv4.execute-api.us-east-1.amazonaws.com/notify-ws"

// const socket = new SockJS(socketURL);
// export const stompClient = Stomp.over(socket);