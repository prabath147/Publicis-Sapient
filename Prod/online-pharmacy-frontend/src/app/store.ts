import { Action, configureStore, ThunkAction } from "@reduxjs/toolkit";
import StoreReducer from "../components/pages/Admin/storeList/StoreSlice";
import SubscriptionReducer from "../components/pages/AdminSubscriptionPack/SubscriptionvSlice";
import UserReducer from "../components/pages/Auth/Login/UserSlice";
import CartReducer from "../components/pages/Cart/CartSlice";
import ManagerReducer from "../components/pages/Manager/ManagerSlice";
import NotificationReducer from "../components/pages/Open/notificationList/NotificationSlice";
// import OrderSlice from "../components/pages/orderDetail/OrderSlice";
// import OrderListSlice from "../components/pages/orderList/OrderListSlice";

export const store = configureStore({
  reducer: {
    storeL: StoreReducer,
    user: UserReducer,
    notification: NotificationReducer,
    viewsubscription: SubscriptionReducer,
    // addsubscription: AddSubcriptionRe,
    cart: CartReducer,
    // order: OrderSlice,
    // orderL: OrderListSlice,
    manager: ManagerReducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false,
    }),
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
