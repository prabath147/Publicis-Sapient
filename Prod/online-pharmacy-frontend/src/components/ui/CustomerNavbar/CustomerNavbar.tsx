import { Burger, Container, createStyles, Group, Header, Image, Paper, Space, Transition } from '@mantine/core';
import { useDisclosure } from '@mantine/hooks';
import { useNavigate } from 'react-router-dom';
import logo from '../../../images/logo.png';
import UserBadge from '../UserBadge/UserBadge';

const HEADER_HEIGHT = 60;

const useStyles = createStyles((theme) => ({
  root: {
    position: 'relative',
    zIndex: 3,
    // width: '100%',
    display: "block"
  },

  dropdown: {
    position: 'absolute',
    top: HEADER_HEIGHT,
    left: 0,
    right: 0,
    zIndex: 0,
    borderTopRightRadius: 0,
    borderTopLeftRadius: 0,
    borderTopWidth: 0,
    overflow: 'hidden',

    [theme.fn.largerThan('sm')]: {
      display: 'none',
    },
  },

  header: {
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'center',
    height: '100%',
    // width: "100%"
  },

  links: {
    [theme.fn.smallerThan('sm')]: {
      display: 'none',
    },
  },

  burger: {
    [theme.fn.largerThan('sm')]: {
      display: 'none',
    },
  },

  link: {
    display: 'block',
    lineHeight: 1,
    padding: '8px 12px',
    borderRadius: theme.radius.sm,
    textDecoration: 'none',
    color: theme.colors.gray[7],
    fontSize: theme.fontSizes.sm,
    fontWeight: 500,

    '&:hover': {
      backgroundColor: theme.colors.gray[0],
    },

    [theme.fn.smallerThan('sm')]: {
      borderRadius: 0,
      padding: theme.spacing.md,
    },
  },

  linkActive: {
    '&, &:hover': {
      backgroundColor: theme.fn.variant({ variant: 'light', color: theme.primaryColor }).background,
      color: theme.fn.variant({ variant: 'light', color: theme.primaryColor }).color,
    },
  },
}));


const links = [
  { label: "Search", link: "/search" },
  { label: "Subscriptions", link: "/register/subscription" },
  { label: "Cart", link: "/cart" },
]

export default function CustomerNavbar() {
  const [opened, { toggle }] = useDisclosure(false);
  const { classes, cx } = useStyles();
  const navigate = useNavigate()

  const items = links.map((link) => (
    <a
      key={link.label}
      href={link.link}
      // className={cx(classes.link, { [classes.linkActive]: active === link.link })}
      className={cx(classes.link)}
      onClick={(event) => {
        event.preventDefault();
        navigate(link.link)
        // close();
      }}
    >
      {link.label}
    </a>
  ));

  return (
    <Container>
      <Header height={HEADER_HEIGHT} className={classes.root}>
        <Container className={classes.header}>
          <Image src={logo} width={120} height={20} fit="contain" />
          <Group spacing={5} className={classes.links}>
            {items}
            {/* <ActionIcon onClick={() => { navigate("/notify"); }}>
              <FontAwesomeIcon icon={faBell} />
            </ActionIcon> */}
            <UserBadge />
          </Group>

          <Burger opened={opened} onClick={toggle} className={classes.burger} size="sm" data-testid="burger-btn" />

          <Transition transition="pop-top-right" duration={200} mounted={opened}>
            {(styles) => (
              <Paper className={classes.dropdown} withBorder style={styles}>
                {items}
                {/* <ActionIcon onClick={() => { navigate("/notify"); }}>
                  <FontAwesomeIcon icon={faBell} />
                </ActionIcon> */}
                <UserBadge />


              </Paper>
            )}
          </Transition>
        </Container>
      </Header>
      <Space h="md" />
    </Container>
  );
}