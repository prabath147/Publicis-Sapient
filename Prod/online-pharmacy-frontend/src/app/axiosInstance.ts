import axios, { AxiosRequestConfig } from "axios";
import LocalUserService from "../services/LocalUserService";
import TokenService from "../services/TokenService";
import { baseURL } from "./baseUrl";

const axiosInstance = axios.create({
  baseURL: baseURL,
  timeout: 5000,
  headers: {
    "Content-Type": "application/json",
    accept: "application/json",
  },
});

axiosInstance.interceptors.request.use(
  (config: AxiosRequestConfig) => {
    const token = TokenService.getLocalAccessToken();
    if (token && config.headers !== undefined) {
      config.headers["Authorization"] = "Bearer " + token; // for Spring Boot back-end
      // config.headers["x-access-token"] = token; // for Node.js Express back-end
      config.headers["ngrok-skip-browser-warning"] = "";
    }
    return config;
  },
  (error) => {
    return Promise.reject(error);
  }
);

axiosInstance.interceptors.response.use(
  (res) => {
    return res;
  },
  async (err) => {
    const originalConfig = err.config;
    if (originalConfig.url !== "api/auth/signin" && err.response) {
      // Access Token was expired
      if (err.response.status === 401 && !originalConfig._retry) {
        originalConfig._retry = true;
        try {
          const rs = await axiosInstance.post("api/auth/refresh-token", {
            refreshToken: TokenService.getLocalRefreshToken(),
          });
          const { accessToken } = rs.data;
          TokenService.updateLocalAccessToken(accessToken);
          return axiosInstance(originalConfig);
        } catch (_error) {
          TokenService.removeLocalAccessToken()
          TokenService.removeLocalRefreshToken()
          LocalUserService.removeLocalUser()
          // window.location.reload();
          return Promise.reject(_error);
        }
      }
    }
    return Promise.reject(err);
  }
);
export default axiosInstance;
