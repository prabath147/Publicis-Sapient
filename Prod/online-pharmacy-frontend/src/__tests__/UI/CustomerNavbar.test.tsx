import { act, fireEvent, render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { store } from '../../app/store';
import CustomerNavbar from '../../components/ui/CustomerNavbar/CustomerNavbar';


describe('CustomerNavbar', () => {

  it('component created', () => {
    const component = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <CustomerNavbar />
        </BrowserRouter>
      </Provider>
    )
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('component works on click event', async () => {
    const { asFragment, getByRole } = render(
      <Provider store={store}>
        <BrowserRouter>
          <CustomerNavbar />
        </BrowserRouter>
      </Provider>
    );

    await act(async () => {
      fireEvent.click(getByRole('link', {
        name: /cart/i
      }))
    })

    expect(asFragment()).toMatchSnapshot();
  })



  it('component works on click event', async () => {
    const { asFragment, getByTestId } = render(
      <div style={{ width: "200px" }}>
        <Provider store={store}>
          <BrowserRouter>
            <CustomerNavbar />
          </BrowserRouter>
        </Provider>
      </div>
    );

    await act(async () => {
      fireEvent.click(getByTestId("burger-btn"))
    })

    expect(asFragment()).toMatchSnapshot();
  })

})