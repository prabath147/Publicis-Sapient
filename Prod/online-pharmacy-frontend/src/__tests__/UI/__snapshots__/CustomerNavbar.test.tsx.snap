// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerNavbar component created 1`] = `
<div
  className="mantine-Container-root mantine-15pjuqq"
>
  <header
    className="mantine-Header-root mantine-1t1et9z"
  >
    <div
      className="mantine-Container-root mantine-1ed2r9l"
    >
      <div
        className="mantine-Image-root mantine-15po0m8"
        style={
          Object {
            "height": 20,
            "width": 120,
          }
        }
      >
        <figure
          className="mantine-11nhzn5 mantine-Image-figure"
        >
          <div
            className="mantine-qqmv3w mantine-Image-imageWrapper"
          >
            <img
              className="mantine-3y8yz3 mantine-Image-image"
              onError={[Function]}
              onLoad={[Function]}
              src="logo.png"
              style={
                Object {
                  "height": 20,
                  "objectFit": "contain",
                  "width": 120,
                }
              }
            />
          </div>
        </figure>
      </div>
      <div
        className="mantine-Group-root mantine-xtpvmj"
      >
        <a
          className="mantine-Group-child mantine-1til4lg"
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
        <a
          className="mantine-Group-child mantine-1til4lg"
          href="/register/subscription"
          onClick={[Function]}
        >
          Subscriptions
        </a>
        <a
          className="mantine-Group-child mantine-1til4lg"
          href="/cart"
          onClick={[Function]}
        >
          Cart
        </a>
        <button
          aria-controls="mantine-r0-dropdown"
          aria-expanded={false}
          aria-haspopup="menu"
          className="mantine-UnstyledButton-root mantine-1ysirc4"
          data-testid="btn-badge"
          id="mantine-r0-target"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          type="button"
        >
          <div
            className="mantine-Avatar-root mantine-1tqi9xs"
          >
            <div
              className="mantine-ttafi4 mantine-Avatar-placeholder"
            >
              <svg
                className="mantine-1vmgx88 mantine-Avatar-placeholderIcon"
                fill="none"
                height="15"
                viewBox="0 0 15 15"
                width="15"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clipRule="evenodd"
                  d="M0.877014 7.49988C0.877014 3.84219 3.84216 0.877045 7.49985 0.877045C11.1575 0.877045 14.1227 3.84219 14.1227 7.49988C14.1227 11.1575 11.1575 14.1227 7.49985 14.1227C3.84216 14.1227 0.877014 11.1575 0.877014 7.49988ZM7.49985 1.82704C4.36683 1.82704 1.82701 4.36686 1.82701 7.49988C1.82701 8.97196 2.38774 10.3131 3.30727 11.3213C4.19074 9.94119 5.73818 9.02499 7.50023 9.02499C9.26206 9.02499 10.8093 9.94097 11.6929 11.3208C12.6121 10.3127 13.1727 8.97172 13.1727 7.49988C13.1727 4.36686 10.6328 1.82704 7.49985 1.82704ZM10.9818 11.9787C10.2839 10.7795 8.9857 9.97499 7.50023 9.97499C6.01458 9.97499 4.71624 10.7797 4.01845 11.9791C4.97952 12.7272 6.18765 13.1727 7.49985 13.1727C8.81227 13.1727 10.0206 12.727 10.9818 11.9787ZM5.14999 6.50487C5.14999 5.207 6.20212 4.15487 7.49999 4.15487C8.79786 4.15487 9.84999 5.207 9.84999 6.50487C9.84999 7.80274 8.79786 8.85487 7.49999 8.85487C6.20212 8.85487 5.14999 7.80274 5.14999 6.50487ZM7.49999 5.10487C6.72679 5.10487 6.09999 5.73167 6.09999 6.50487C6.09999 7.27807 6.72679 7.90487 7.49999 7.90487C8.27319 7.90487 8.89999 7.27807 8.89999 6.50487C8.89999 5.73167 8.27319 5.10487 7.49999 5.10487Z"
                  fill="currentColor"
                  fillRule="evenodd"
                />
              </svg>
            </div>
          </div>
        </button>
      </div>
      <button
        className="mantine-UnstyledButton-root mantine-Burger-root mantine-16hx9h7"
        data-testid="burger-btn"
        onClick={[Function]}
        type="button"
      >
        <div
          className="mantine-1hfrgwx mantine-Burger-burger"
        />
      </button>
    </div>
  </header>
  <div
    className="mantine-1ne3752"
  />
</div>
`;

exports[`CustomerNavbar component works on click event 1`] = `
<DocumentFragment>
  <div
    class="mantine-Container-root mantine-15pjuqq"
  >
    <header
      class="mantine-Header-root mantine-1t1et9z"
    >
      <div
        class="mantine-Container-root mantine-1ed2r9l"
      >
        <div
          class="mantine-Image-root mantine-15po0m8"
          style="width: 120px; height: 20px;"
        >
          <figure
            class="mantine-11nhzn5 mantine-Image-figure"
          >
            <div
              class="mantine-qqmv3w mantine-Image-imageWrapper"
            >
              <img
                class="mantine-3y8yz3 mantine-Image-image"
                src="logo.png"
                style="object-fit: contain; width: 120px; height: 20px;"
              />
            </div>
          </figure>
        </div>
        <div
          class="mantine-Group-root mantine-xtpvmj"
        >
          <a
            class="mantine-Group-child mantine-1til4lg"
            href="/search"
          >
            Search
          </a>
          <a
            class="mantine-Group-child mantine-1til4lg"
            href="/register/subscription"
          >
            Subscriptions
          </a>
          <a
            class="mantine-Group-child mantine-1til4lg"
            href="/cart"
          >
            Cart
          </a>
          <button
            aria-controls="mantine-r0-dropdown"
            aria-expanded="false"
            aria-haspopup="menu"
            class="mantine-UnstyledButton-root mantine-1ysirc4"
            data-testid="btn-badge"
            id="mantine-r0-target"
            type="button"
          >
            <div
              class="mantine-Avatar-root mantine-1tqi9xs"
            >
              <div
                class="mantine-ttafi4 mantine-Avatar-placeholder"
              >
                <svg
                  class="mantine-1vmgx88 mantine-Avatar-placeholderIcon"
                  fill="none"
                  height="15"
                  viewBox="0 0 15 15"
                  width="15"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M0.877014 7.49988C0.877014 3.84219 3.84216 0.877045 7.49985 0.877045C11.1575 0.877045 14.1227 3.84219 14.1227 7.49988C14.1227 11.1575 11.1575 14.1227 7.49985 14.1227C3.84216 14.1227 0.877014 11.1575 0.877014 7.49988ZM7.49985 1.82704C4.36683 1.82704 1.82701 4.36686 1.82701 7.49988C1.82701 8.97196 2.38774 10.3131 3.30727 11.3213C4.19074 9.94119 5.73818 9.02499 7.50023 9.02499C9.26206 9.02499 10.8093 9.94097 11.6929 11.3208C12.6121 10.3127 13.1727 8.97172 13.1727 7.49988C13.1727 4.36686 10.6328 1.82704 7.49985 1.82704ZM10.9818 11.9787C10.2839 10.7795 8.9857 9.97499 7.50023 9.97499C6.01458 9.97499 4.71624 10.7797 4.01845 11.9791C4.97952 12.7272 6.18765 13.1727 7.49985 13.1727C8.81227 13.1727 10.0206 12.727 10.9818 11.9787ZM5.14999 6.50487C5.14999 5.207 6.20212 4.15487 7.49999 4.15487C8.79786 4.15487 9.84999 5.207 9.84999 6.50487C9.84999 7.80274 8.79786 8.85487 7.49999 8.85487C6.20212 8.85487 5.14999 7.80274 5.14999 6.50487ZM7.49999 5.10487C6.72679 5.10487 6.09999 5.73167 6.09999 6.50487C6.09999 7.27807 6.72679 7.90487 7.49999 7.90487C8.27319 7.90487 8.89999 7.27807 8.89999 6.50487C8.89999 5.73167 8.27319 5.10487 7.49999 5.10487Z"
                    fill="currentColor"
                    fill-rule="evenodd"
                  />
                </svg>
              </div>
            </div>
          </button>
        </div>
        <button
          class="mantine-UnstyledButton-root mantine-Burger-root mantine-16hx9h7"
          data-testid="burger-btn"
          type="button"
        >
          <div
            class="mantine-1hfrgwx mantine-Burger-burger"
          />
        </button>
      </div>
    </header>
    <div
      class="mantine-1ne3752"
    />
  </div>
</DocumentFragment>
`;

exports[`CustomerNavbar component works on click event 2`] = `
<DocumentFragment>
  <div
    style="width: 200px;"
  >
    <div
      class="mantine-Container-root mantine-15pjuqq"
    >
      <header
        class="mantine-Header-root mantine-1t1et9z"
      >
        <div
          class="mantine-Container-root mantine-1ed2r9l"
        >
          <div
            class="mantine-Image-root mantine-15po0m8"
            style="width: 120px; height: 20px;"
          >
            <figure
              class="mantine-11nhzn5 mantine-Image-figure"
            >
              <div
                class="mantine-qqmv3w mantine-Image-imageWrapper"
              >
                <img
                  class="mantine-3y8yz3 mantine-Image-image"
                  src="logo.png"
                  style="object-fit: contain; width: 120px; height: 20px;"
                />
              </div>
            </figure>
          </div>
          <div
            class="mantine-Group-root mantine-xtpvmj"
          >
            <a
              class="mantine-Group-child mantine-1til4lg"
              href="/search"
            >
              Search
            </a>
            <a
              class="mantine-Group-child mantine-1til4lg"
              href="/register/subscription"
            >
              Subscriptions
            </a>
            <a
              class="mantine-Group-child mantine-1til4lg"
              href="/cart"
            >
              Cart
            </a>
            <button
              aria-controls="mantine-r1-dropdown"
              aria-expanded="false"
              aria-haspopup="menu"
              class="mantine-UnstyledButton-root mantine-1ysirc4"
              data-testid="btn-badge"
              id="mantine-r1-target"
              type="button"
            >
              <div
                class="mantine-Avatar-root mantine-1tqi9xs"
              >
                <div
                  class="mantine-ttafi4 mantine-Avatar-placeholder"
                >
                  <svg
                    class="mantine-1vmgx88 mantine-Avatar-placeholderIcon"
                    fill="none"
                    height="15"
                    viewBox="0 0 15 15"
                    width="15"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M0.877014 7.49988C0.877014 3.84219 3.84216 0.877045 7.49985 0.877045C11.1575 0.877045 14.1227 3.84219 14.1227 7.49988C14.1227 11.1575 11.1575 14.1227 7.49985 14.1227C3.84216 14.1227 0.877014 11.1575 0.877014 7.49988ZM7.49985 1.82704C4.36683 1.82704 1.82701 4.36686 1.82701 7.49988C1.82701 8.97196 2.38774 10.3131 3.30727 11.3213C4.19074 9.94119 5.73818 9.02499 7.50023 9.02499C9.26206 9.02499 10.8093 9.94097 11.6929 11.3208C12.6121 10.3127 13.1727 8.97172 13.1727 7.49988C13.1727 4.36686 10.6328 1.82704 7.49985 1.82704ZM10.9818 11.9787C10.2839 10.7795 8.9857 9.97499 7.50023 9.97499C6.01458 9.97499 4.71624 10.7797 4.01845 11.9791C4.97952 12.7272 6.18765 13.1727 7.49985 13.1727C8.81227 13.1727 10.0206 12.727 10.9818 11.9787ZM5.14999 6.50487C5.14999 5.207 6.20212 4.15487 7.49999 4.15487C8.79786 4.15487 9.84999 5.207 9.84999 6.50487C9.84999 7.80274 8.79786 8.85487 7.49999 8.85487C6.20212 8.85487 5.14999 7.80274 5.14999 6.50487ZM7.49999 5.10487C6.72679 5.10487 6.09999 5.73167 6.09999 6.50487C6.09999 7.27807 6.72679 7.90487 7.49999 7.90487C8.27319 7.90487 8.89999 7.27807 8.89999 6.50487C8.89999 5.73167 8.27319 5.10487 7.49999 5.10487Z"
                      fill="currentColor"
                      fill-rule="evenodd"
                    />
                  </svg>
                </div>
              </div>
            </button>
          </div>
          <button
            class="mantine-UnstyledButton-root mantine-Burger-root mantine-16hx9h7"
            data-testid="burger-btn"
            type="button"
          >
            <div
              class="mantine-1hfrgwx mantine-Burger-burger"
              data-opened="true"
            />
          </button>
          <div
            class="mantine-Paper-root mantine-18luckg"
            style="transition-property: transform, opacity; transition-duration: 200ms; transition-timing-function: ease; transform-origin: top right; opacity: 1; transform: scale(1);"
          >
            <a
              class="mantine-26uxky"
              href="/search"
            >
              Search
            </a>
            <a
              class="mantine-26uxky"
              href="/register/subscription"
            >
              Subscriptions
            </a>
            <a
              class="mantine-26uxky"
              href="/cart"
            >
              Cart
            </a>
            <button
              aria-controls="mantine-r2-dropdown"
              aria-expanded="false"
              aria-haspopup="menu"
              class="mantine-UnstyledButton-root mantine-1ysirc4"
              data-testid="btn-badge"
              id="mantine-r2-target"
              type="button"
            >
              <div
                class="mantine-Avatar-root mantine-1tqi9xs"
              >
                <div
                  class="mantine-ttafi4 mantine-Avatar-placeholder"
                >
                  <svg
                    class="mantine-1vmgx88 mantine-Avatar-placeholderIcon"
                    fill="none"
                    height="15"
                    viewBox="0 0 15 15"
                    width="15"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M0.877014 7.49988C0.877014 3.84219 3.84216 0.877045 7.49985 0.877045C11.1575 0.877045 14.1227 3.84219 14.1227 7.49988C14.1227 11.1575 11.1575 14.1227 7.49985 14.1227C3.84216 14.1227 0.877014 11.1575 0.877014 7.49988ZM7.49985 1.82704C4.36683 1.82704 1.82701 4.36686 1.82701 7.49988C1.82701 8.97196 2.38774 10.3131 3.30727 11.3213C4.19074 9.94119 5.73818 9.02499 7.50023 9.02499C9.26206 9.02499 10.8093 9.94097 11.6929 11.3208C12.6121 10.3127 13.1727 8.97172 13.1727 7.49988C13.1727 4.36686 10.6328 1.82704 7.49985 1.82704ZM10.9818 11.9787C10.2839 10.7795 8.9857 9.97499 7.50023 9.97499C6.01458 9.97499 4.71624 10.7797 4.01845 11.9791C4.97952 12.7272 6.18765 13.1727 7.49985 13.1727C8.81227 13.1727 10.0206 12.727 10.9818 11.9787ZM5.14999 6.50487C5.14999 5.207 6.20212 4.15487 7.49999 4.15487C8.79786 4.15487 9.84999 5.207 9.84999 6.50487C9.84999 7.80274 8.79786 8.85487 7.49999 8.85487C6.20212 8.85487 5.14999 7.80274 5.14999 6.50487ZM7.49999 5.10487C6.72679 5.10487 6.09999 5.73167 6.09999 6.50487C6.09999 7.27807 6.72679 7.90487 7.49999 7.90487C8.27319 7.90487 8.89999 7.27807 8.89999 6.50487C8.89999 5.73167 8.27319 5.10487 7.49999 5.10487Z"
                      fill="currentColor"
                      fill-rule="evenodd"
                    />
                  </svg>
                </div>
              </div>
            </button>
          </div>
        </div>
      </header>
      <div
        class="mantine-1ne3752"
      />
    </div>
  </div>
</DocumentFragment>
`;
